vpr/src/place/place.cpp

```diff
@@ -670,30 +793,64 @@ void try_place(const t_placer_opts& placer_opts,
                                       pin_timing_invalidator.get(),
                                       timing_info.get());
 
-        placement_inner_loop(&state, placer_opts,
-                             inner_recompute_limit, &stats,
-                             &costs,
-                             &moves_since_cost_recompute,
-                             pin_timing_invalidator.get(),
-                             place_delay_model.get(),
-                             placer_criticalities.get(),
-                             placer_setup_slacks.get(),
-                             *move_generator,
-                             blocks_affected,
-                             timing_info.get(),
-                             placer_opts.place_algorithm);
-
-        tot_iter += state.move_lim;
-        ++state.num_temps;
-
         if (placer_opts.place_algorithm.is_timing_driven()) {
             critical_path = timing_info->least_slack_critical_path();
             sTNS = timing_info->setup_total_negative_slack();
             sWNS = timing_info->setup_worst_negative_slack();
+
+            //see if we should save the current placement solution as a checkpoint
+            if (placer_opts.place_checkpointing && agent_state == 2) {
```

Comment:
Not sure what agent_state == 2 means.  Use enum.

This would be a good spot for a helper function:  checkpoint_placement (...) could do all this.

Reply:
solved by defining enum and implementing a new helper function to check whether we need to save placement checkpoint or not.